src/zoid/buttons/util.js

```diff
@@ -82,9 +82,35 @@ export function applePaySession() : ?ApplePaySessionConfigRequest {
 
         return (version, request) => {
             const session = new window.ApplePaySession(version, request);
+            const listeners = {};
+
+            session.onvalidatemerchant = (e) => {
+                listeners.validatemerchant(e.validationURL);
```

Comment:
Can we keep the same structure of the event object? e.g. `listeners.validatemerchant({ validationURL: e.validationURL })`